packages/fxa-auth-server/scripts/paypal-processor.ts

```diff
@@ -65,5 +121,6 @@ if (require.main === module) {
       console.error(err);
       process.exit(1);
     })
-    .then((result) => process.exit(result));
+    .then((result) => process.exit(result))
+    .finally(() => lock?.release());
```

Comment:
Q: Will a `finally()` run after a `process.exit()`? or is `process.exit()` immediate?
(although that should be a quick experiment for me to run in isolation separately)

Reply:
hah yeah, I think you are right that that `finally` won't get executed.  I'll fix.  Thanks!

Reply:
Doesn't look like it:

```js
Promise.resolve("OK, computer")
  .then(msg => {
    console.log("success:", msg);
    throw new Error("Ruh roh");
  })
  .catch(err => {
    console.error("error:", err.message);
    process.exit(1);
  })
  .finally(() => console.log("I'm still here..."));
```

```sh
node --version # v16.14.2
node index.mjs
# success: OK, computer
# error: Ruh roh
```

Note that the `.finally()` log never seems to fire due to the hard `process.exit()`.

Reply:
Deeper and deeper&hellip; This seems to do what I'd expect, but not sure if I know what I'm doing:

```js
async function init(e=false) {
  if (e) {
    throw new Error("OH NO!");
  }
  return 0;
}

init(false)
  .then((result) => {
    console.log("then...", result);
    process.exitCode = result;
  })
  .catch((err) => {
    console.error("catch...", err.message);
    process.exitCode = 1;
  })
  .finally(() => {
    console.log("finally...", process.exitCode);
    process.exit(process.exitCode);
  });
```

I had to swap the `.then()` and `.catch()` order, since `.catch()` swallows errors then proceeds onto the next `.then()` chain.

But now if the initial `init()` _or_ `.then()` block fails, the code should go into the `.catch()` block and set a non-zero exit code which falls through to the `.finally()` block which can release locks and set hard exit with your desired exit code.